apps/wing/src/analytics/storage.ts

```diff
@@ -0,0 +1,94 @@
+import { readFileSync, writeFileSync, mkdirSync, existsSync } from "fs";
+import { AnalyticEvent } from "./event";
+import { randomBytes } from "crypto";
+import {v4 as uuidv4 } from "uuid";
+import path from "path";
+import * as os from "os";
+
+const WING_HOME_DIR = path.join(os.homedir(), ".wing")
+const WING_ANALYTICS_CONFIG_FILE = path.join(WING_HOME_DIR, 'wing-analytics-config.json');
+
+interface AnalyticsConfig {
+  anonymousId: string;
+  disclaimerDisplayed?: boolean;
+  optOut?: boolean;
+}
+
+/**
+ * Stores an analytic event to disk for later export
+ * 
+ * @param event the event to store
+ * @returns string the file path of the stored analytic
+ */
+export function storeAnalyticEvent(event: AnalyticEvent): string {
+  const eventId = uuidv4();
+  const tmpdir = path.join(os.tmpdir(), "wing-analytics");
+  
+  if (!existsSync(tmpdir)) {
+    mkdirSync(tmpdir);
+  }
```

Comment:
i think we *never* want to fail if something doesn't work with our analytics. probably need to surround everything with try/catch like you did in the `detached-export` script

Reply:
For this particular chunk, the exception is actually used for logic rather than just handling failures. But yeah I would agree this whole thing also needs a try/catch anyways.